src/bestcrypt_fmt_plug.c

```diff
@@ -0,0 +1,494 @@
+	// container-password@123.jbc, second active key slot
+	{"$BestCrypt$1$5$3$16384$240$3154116612$129$64$c8d1e54303eb356636e138295e49533f250d4c327f57ceae134cb3995a7c01ea8814b0352dd7300899d2adf803c90f7a6872c27234693a8ba2dd354be30dbee5$1$4c72ff9af13e578d116d48e251e93a87c43cb518e0256b0092c9db1dabe7c1422105423a1a7a8fb21c7e8c22e4542a9d58b3bc17198a9ec1151ac3e7d20cf5db9c24c39ec3f259a47cac4e6340b42d4b025942796f616ed088c162bdd78ef0cce258ec5e5b2c0b5915d0bfd5432b578a3694cefff2a9a4a3e1367279fcde9c7c9575f945ce704b888089029df99d4613b1fa8f4a16bd18bfc346942eda549bf4ff216cd6f304f284dea1a3e1a10fe6f7b74b3661980a5b8cd8fbe8a00b2065abb23822477caa88c4881e7bebefff696c1e205213800066fd4f28b56abb36585f51b37de4ffd7f2b6597db33ebb60d70ce00a38d0a4ba549a15d97a780986bed6", "openwall123"},
+	{NULL}
+};
+
+typedef struct {
+        unsigned char data[kKeySlotSize];
+} KEY_BLOCK;
+
+// PBE key block format for version 5
+typedef struct {
+	unsigned char keyblock[128]; // KGEncryptedBlock32<key_data_t> keyblock, encoded key, is this size fixed?
+	unsigned char iv[kIVSize]; // unencrypted initial vector
+} pbe_format_v5_32; // used when digestSize <= 32
+
+typedef struct {
+	unsigned char keyblock[160]; // KGEncryptedBlock64<key_data_t> keyblock, encoded key, is this size fixed?
+	unsigned char iv[kIVSize]; // unencrypted initial vector
+} pbe_format_v5_64; // used when digestSize <= 64
+
+struct KGEncryptedBlock32 {
+	unsigned char data[96]; // is this offset fixed?
+	unsigned char digest[kDigestSize32];
+};
+
+struct KGEncryptedBlock64 {
+	unsigned char data[96]; // is this offset fixed?
+	unsigned char digest[kDigestSize32];
+};
+
+static struct custom_salt {
+	int version;
+	int wKeyGenId;
+	int wVersion;
+	int iterations;
+	int alg_id;
+	uint32_t mode_id;
+	int hash_id;
+	int salt_size;
+	unsigned char salt[64];
+	int active_slots;
+	unsigned char key[256]; // one active key per hash
+} *cur_salt;
+
+static char (*saved_key)[PLAINTEXT_LENGTH + 2];
+static int *saved_len;
+static int *cracked, cracked_count;
+
+static void init(struct fmt_main *self)
+{
+#ifdef _OPENMP
+	static int omp_t = 1;
+	omp_t = omp_get_max_threads();
+	self->params.min_keys_per_crypt *= omp_t;
+	omp_t *= OMP_SCALE;
+	self->params.max_keys_per_crypt *= omp_t;
+#endif
+	saved_key = mem_calloc(sizeof(*saved_key), self->params.max_keys_per_crypt);
+	saved_len = mem_calloc(self->params.max_keys_per_crypt, sizeof(*saved_len));
+	cracked = mem_calloc(sizeof(*cracked), self->params.max_keys_per_crypt);
+	cracked_count = self->params.max_keys_per_crypt;
+}
+
+static void done(void)
+{
+	MEM_FREE(cracked);
+	MEM_FREE(saved_key);
+}
+
+static int valid(char *ciphertext, struct fmt_main *self)
+{
+	char *p = ciphertext, *ctcopy, *keeptr;
+	int salt_size, extra;
+	int res;
+	uint32_t mode_id;
+
+	if (strncasecmp(ciphertext, FORMAT_TAG, FORMAT_TAG_LENGTH))
```

Comment:
We should not introduce case insensitive format tags, especially when we have a bestcrypt2john utility that creates the input.